File: src/test/java/java_bootcamp/StateTests.java
package java_bootcamp;

import net.corda.core.contracts.ContractState;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.identity.Party;
import net.corda.testing.core.TestIdentity;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class StateTests {
    private final Party alice = new TestIdentity(new CordaX500Name("Alice", "", "GB")).getParty();
    private final Party bob = new TestIdentity(new CordaX500Name("Bob", "", "GB")).getParty();

//    @Test
//    public void tokenStateHasIssuerOwnerAndAmountFieldsOfTheCorrectType() {
//        new TokenState(alice, bob, 1);
//    }

//    @Test
//    public void tokenStateHasGettersForIssuerOwnerAndAmount() {
//        TokenState tokenState = new TokenState(alice, bob, 1);
//        assertEquals(alice, tokenState.getIssuer());
//        assertEquals(bob, tokenState.getOwner());
//        assertEquals(1, tokenState.getAmount());
//    }

//    @Test
//    public void tokenStateImplementsContractState() {
//        assert(new TokenState(alice, bob, 1) instanceof ContractState);
//    }

//    @Test
//    public void tokenStateHasTwoParticipantsTheIssuerAndTheOwner() {
//        TokenState tokenState = new TokenState(alice, bob, 1);
//        assertEquals(2, tokenState.getParticipants().size());
//        assert(tokenState.getParticipants().contains(alice));
//        assert(tokenState.getParticipants().contains(bob));
//    }
}

File: src/test/kotlin/bootcamp/StateTests.kt
package bootcamp

import net.corda.core.contracts.ContractState
import net.corda.core.identity.CordaX500Name
import net.corda.testing.core.TestIdentity
import org.junit.Test
import kotlin.test.assertEquals

class StateTests {
    private val alice = TestIdentity(CordaX500Name("Alice", "", "GB")).party
    private val bob = TestIdentity(CordaX500Name("Bob", "", "GB")).party

    @Test
    fun `TokenState has issuer, recipient and amount fields of the correct type`() {
        TokenState(alice, bob, 1)
    }

    @Test
    fun `TokenState implements ContractState`() {
        assert(TokenState(alice, bob, 1) is ContractState)
    }

    @Test
    fun `TokenState has two participants, the issuer and the recipient`() {
        val tokenState = TokenState(alice, bob, 1)
        assertEquals(setOf(alice, bob), tokenState.participants.toSet())
    }
}